### PR TITLE
Add Xenia (Xbox 360) as a game source

### PR DESCRIPTION
**Pull request by Salt-555**
## Summary

Adds Xbox 360 game discovery and launching through [Xenia Canary](https://github.com/xenia-canary/xenia-canary), following the established emulator source pattern (Cemu, shadPS4, Dolphin).

## What it does

- **Scanner** (`src/sources/xenia/`): reads `recent.toml` from Xenia storage roots (`~/.local/share/Xenia`, `~/Documents/Xenia`, portable installs), normalizes Wine `Z:\...` paths written by Proton-based setups, and scans the storage root for `default.xex` dumps and ISO/XEX images with sidecar cover art.
- **Game model** (`src/library/XeniaGameModel`): SQLite cache (`xenia_games`), async scanning, favorite/hidden persistence, same roles/system taxonomy as the other console sources (`xbox360`).
- **Launcher**: validates `iso`/`xex` targets and resolves `xenia_canary` (plus common aliases) from PATH.
- **UI**: settings panel row with detection status, library filter chip with controller navigation, couch-mode filter, game-details manage button, empty/error states.
- **Backups**: `xenia_games` personal flags captured and restored with the existing round-trip.
- **Auto-detection**: mirrors Cemu/shadPS4 — the source enables itself when Xenia is detected and the user has not made an explicit choice.

## One behavioral note for review

The scanner's highest-confidence source is `recent.toml`, which Xenia only writes for games that have actually been launched at least once. A fresh Xenia install with no runs yet will show no games until the user launches one (or places dumps in the storage root, which is also scanned). If you'd prefer richer discovery (e.g. parsing `title_path`-style config entries or exposing a manual folder picker), happy to iterate.

## Testing

- `omakade_core_tests`: 156 passed, 0 failed
- `omakade_backup_recovery_tests`: 21 passed, 0 failed (includes a new `xenia_games` fixture row in the round-trip test)
- Built clean with `cmake --preset dev && cmake --build --preset dev`

## Checklist

- [x] Pattern-matched against Cemu/shadPS4/Dolphin sources (record structs, scanner API, model lifecycle, launcher validation, QML wiring, backup snapshot)
- [x] Path validation reuses `validLaunchPath` guards (no shell invocation; QProcess argument lists)
- [x] Config keys follow the `*_enabled` / auto-detection convention
- [x] Docs updated (`docs/BACKUP-FORMAT.md`)

## Summary by CodeRabbit

* **New Features**
  * Added Xbox 360 game support through Xenia Canary.
  * Discover games from recent titles, storage folders, XEX files, ISO images, and ZAR archives.
  * Browse, favorite, hide, track play sessions, and launch Xenia games.
  * Added Xenia filtering, settings, detection status, error messages, refresh, and rescanning controls.
  * Included Xenia games in library backups and the Emulated source filter.

* **Documentation**
  * Added Xenia support details to the README and unreleased changelog.
  * Documented Xenia backup data handling.

### CODE REVIEW

**Comment by btsouth** [comment]
Thanks for adding Xenia support! Could you take a look at the failing CI checks and the existing review feedback, and update the branch against main? Happy to give it another pass and test locally once it's updated.

**Review by btsouth — DISMISSED** [review]
Thanks again for the contribution. Before local Xenia testing, this needs an update onto current main, green deterministic launcher tests, and fixes for inaccessible storage and Windows-path handling from the existing review. Please ping when it’s ready for another pass.

**Comment by btsouth** [comment]
Thanks again for this. We picked the branch up rather than let it stall: it is rebased onto current main, all checks are green, and the review findings are addressed. Your original commit is unchanged and still authored by you.

What changed on top of your work:

- Rebased onto main and kept Xenia alongside the new RomM source, save management, and the reworked sources menu.
- Made the launcher test deterministic with a private PATH and added a missing-emulator case, which clears the CI failure.
- The launcher now accepts ZAR images to match what the scanner imports.
- normalizeWindowsPath only rewrites Wine Z: paths. Other drive letters are skipped with a warning instead of being imported as host paths that do not exist.
- recent.toml parsing handles both quote styles and any key order.
- Scans are marked incomplete when a root or a recent game is unreadable, so the cached library is preserved instead of being hidden.
- Playtime and last-played now come from the session recorder like the other emulator sources.
- Dropped the org.xenia.xenia discovery root since there is no official Xenia Flatpak.

Tests: full suite is green (246/246 locally, both architecture CI jobs pass). No real Xenia install was available here, so launch and return still need a maintainer pass before merge.

File: src/sources/xenia/XeniaScanner.cpp [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+    // appear in saved paths). Drive prefix and leading backslash come off,
+    // then separators flip to forward slashes.
+    if (path->length() >= 3 && (*path)[2] == QLatin1Char('\\')) {
+      path->remove(0, 3);
+    } else {
+      path->remove(0, 2);
+    }
+    path->replace(QLatin1Char('\\'), QLatin1Char('/'));
+    path->prepend(QLatin1Char('/'));
+  }
```
[the remark this answers is not in the document]
Comment (btsouth) [reply src/sources/xenia/XeniaScanner.cpp:102]: Only Wine Z: is rewritten now. Other drive letters are skipped with a warning instead of being imported as host paths that do not exist.

File: src/sources/xenia/XeniaScanner.cpp [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+      }
+      if (title.isEmpty()) {
+        continue;
+      }
+      result.games.append(XeniaGameRecord{.gameId = gameId,
+                                          .title = title,
+                                          .path = recent.path,
+                                          .coverPath = sidecarCover(recent.path),
+                                          .flatpak = false,
+                                          .flatpakAppId = QStringLiteral("")});
```
[the remark this answers is not in the document]
Comment (btsouth) [reply src/sources/xenia/XeniaScanner.cpp:352]: Removed the org.xenia.xenia root because there is no official Xenia Flatpak, so there is no Flatpak state to propagate or a launch path to add.

File: tests/CoreTests.cpp [hunk 3]
```diff
@@ -5815,6 +5821,93 @@ void CoreTests::malformedCemuDataDoesNotReplaceCachedGames() {
...
+  QCOMPARE(model.data(model.index(0), GameRoles::Source).toString(), QStringLiteral("Xenia"));
+  QCOMPARE(model.data(model.index(0), GameRoles::System).toString(), QStringLiteral("xbox360"));
+  model.toggleFavorite(0);
+  model.refreshFromRoots({root});
+  QVERIFY(model.data(model.index(0), GameRoles::Favorite).toBool());
+}
+
+void CoreTests::xeniaLauncherBuildsSafeCommands() {
+  const LaunchCommand command =
+      GameLauncher::xeniaCommand(QStringLiteral("/games/Fable II/default.xex"));
```
[the remark this answers is not in the document]
Comment (btsouth) [reply tests/CoreTests.cpp:6175]: The test now creates a temp executable on a private PATH and a separate missing-emulator test covers the empty-command case. CI is green.

File: src/library/XeniaGameModel.cpp [hunk 4]
```diff
@@ -0,0 +1,323 @@
...
+                               QObject* parent)
+    : QAbstractListModel(parent),
+      m_connectionName(QStringLiteral("omakade-xenia-%1").arg(reinterpret_cast<quintptr>(this))),
+      m_playSessions(playSessions) {
+  if (m_playSessions != nullptr) {
+    connect(m_playSessions, &PlaySessionStore::totalsChanged, this, [this] {
+      if (!m_games.isEmpty()) {
+        emit dataChanged(index(0), index(static_cast<int>(m_games.size()) - 1),
+                         {GameRoles::Hours, GameRoles::PlaytimeSeconds, GameRoles::PlaytimeText,
+                          GameRoles::PlaytimeProvenance, GameRoles::LastPlayed});
```
[the remark this answers is not in the document]
Comment (btsouth) [reply src/library/XeniaGameModel.cpp]: Added GameRoles::Recent to the play-session dataChanged roles.

File: src/sources/xenia/XeniaScanner.cpp [hunk 5]
```diff
@@ -0,0 +1,353 @@
...
+    return;
+  }
+  const QDir dir(directory);
+  if (!dir.exists()) {
+    return;
+  }
+  if (!QFileInfo(directory).isReadable()) {
+    result->incomplete = true;
+    result->warnings.append(QStringLiteral("Could not read %1").arg(directory));
+    return;
```
[the remark this answers is not in the document]
Comment (btsouth) [reply src/sources/xenia/XeniaScanner.cpp:198]: collectGames now checks each candidate target (default.xex, ISO, XEX, ZAR) for readability, warns, marks the scan incomplete, and skips it.
